FsSql/AsyncExtensions.fs

```diff
@@ -4,44 +4,25 @@ open System
 open System.Collections.Generic
 open System.Data
 open System.Data.SqlClient
+open System.Threading
+open System.Threading.Tasks
 open System.Xml
 
 type SqlCommand with
-    member cmd.AsyncExecNonQuery() =
-        let abegin (a,b) = cmd.BeginExecuteNonQuery(a,b)
-        let aend = cmd.EndExecuteNonQuery
-        let acancel = cmd.Cancel
-        Async.FromBeginEnd(abegin, aend, acancel)
+    member cmd.AsyncExecNonQuery(?cancellationToken) =
+        match cancellationToken with
+        | Some ct -> cmd.ExecuteNonQueryAsync(ct) |> Async.AwaitTask
+        | None    -> cmd.ExecuteNonQueryAsync() |> Async.AwaitTask
 
-    member cmd.AsyncExecReader() =
-        let abegin (a,b) = cmd.BeginExecuteReader(a,b)
-        let aend = cmd.EndExecuteReader
-        let acancel = cmd.Cancel
-        Async.FromBeginEnd(abegin, aend, acancel)
+    member cmd.AsyncExecReader(?cancellationToken:CancellationToken) : SqlDataReader Async =
+        match cancellationToken with
+        | Some ct -> cmd.ExecuteReaderAsync(ct) |> Async.AwaitTask
```

Comment:
This is bugged since async needs to be re-executable (cold) and tasks are hot. See if you can defer the creation of the task into an async workflow or something of the like.

Reply:
Ahh, got it! How can I update the PR?